### PR TITLE
2882.1 update b/e crates

### PR DESCRIPTION
**Pull request by roxy-dao**
Part of #2882

# 👩🏻‍💻 What does this PR do?
Updates all the minor/patch crates. There are 5 major updates left coming up soon...

## 💌 Any notes for the reviewer?

# 🧪 Testing

- [ ] Make sure tests are still running okay and everything is still working alright.

# 📃 Documentation

- [ ] **Part of an epic**: documentation will be completed for the feature as a whole
- [x] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [ ] **These areas should be updated or checked**: 
  1.
  2.

### CODE REVIEW

File: server/server/src/certs.rs [hunk 1]
```diff
@@ -118,11 +118,11 @@ impl Certificates {
 
         let key_file = cert_dir.join(PRIVATE_CERT_FILE);
         let mut file = std::fs::File::create(&key_file)?;
-        file.write_all(cert.serialize_private_key_pem().as_bytes())?;
+        file.write_all(cert.cert.pem().as_bytes())?;
```
Comment (roxy-dao) [inline server/server/src/certs.rs]: Now handled by KeyPair::serialize_pem()

File: server/server/src/certs.rs [hunk 2]
```diff
@@ -118,11 +118,11 @@ impl Certificates {
...
 
         let cert_file = cert_dir.join(PUBLIC_CERT_FILE);
         let mut file = std::fs::File::create(&cert_file)?;
-        let public_cert_buffer = cert.serialize_pem()?;
+        let public_cert_buffer = cert.cert.pem();
```
Comment (roxy-dao) [inline server/server/src/certs.rs:126]: Now serialised with Certificate::pem

File: server/Cargo.toml [hunk 3]
```diff
@@ -3,28 +3,28 @@ strip = true
...
+chrono = { version = "0.4.38", features = ["serde"] }
 reqwest = { version = "0.12", features = [
   "json",
   "rustls-tls",
   "multipart",
   "blocking",
 ], default-features = false }
-serde = "1.0.126"
-serde_json = "1.0.66"
+serde = "1.0.203"
```
Comment (jmbrunskill) [inline server/Cargo.toml:22]:
```suggestion
serde = "1.0"
```
Can't we just do this, and always have the latest 1.0 version?

File: server/android/Cargo.toml [hunk 4]
```diff
@@ -9,21 +9,21 @@ path = "src/lib.rs"
...
+rcgen = "0.13.1"
 repository = { path = "../repository" }
 service = { path = "../service" }
 server = { path = "../server", default-features = false, features = [
     "android",
 ] }
-tokio = { version = "1.17.0", features = ["macros"] }
-android_logger = { version = "0.13.0" }
-simple-log = { version = "1.6" }
+tokio = { version = "1.38.0", features = ["macros"] }
```
Comment (jmbrunskill) [inline server/android/Cargo.toml:24]: Tokio and other depenancies are used in lots of places, can we move them to be workspace?
E.g. 
`tokio = { workspace = true }`

Then we should mostly only need to update our workspace versions?
Personally I'd try to move as many as possible to this approach. Should help with version consistency across our crates...

**Review by jmbrunskill — APPROVED** [review]
Looks good to me,
I haven't tested as I don't want to download all the new crates on my phone data haha.

I think we should consider putting more of this into the workspace level to avoid making the same changes in multiple crates?

I'm also not sure if we gain anything when going from say `1.0` to `1.0.2` in theory we should already get the `1.0.2` version when the lock file is recalculated right? And doesn't require further manual intervention...

File: server/Cargo.toml (hunk 3 above)
Comment (roxy-dao) [reply server/Cargo.toml:22]: Oh yes can!

File: server/android/Cargo.toml (hunk 4 above)
Comment (roxy-dao) [reply server/android/Cargo.toml:24]: Can do

Comment (jmbrunskill) [reply server/android/Cargo.toml:24]: Thanks!

**Pull request merged**